app/metal-controller-manager/api/v1alpha1/serverclass_filter.go

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1
+
+import "sort"
+
+// FilterAcceptedServers returns a new slice of Servers that are accepted and qualify.
+//
+// Returned Servers are always sorted by name for stable results.
+func FilterAcceptedServers(servers []Server, q Qualifiers) []Server {
+	servers = filterAccepted(servers)
+	servers = filterCPU(servers, q.CPU)
+	servers = filterSysInfo(servers, q.SystemInformation)
+	servers = filterLabels(servers, q.LabelSelectors)
+
+	sort.Slice(servers, func(i, j int) bool { return servers[i].Name < servers[j].Name })
+
+	return servers
+}
+
+func filterAccepted(servers []Server) []Server {
+	res := make([]Server, 0, len(servers))
+
+	for _, server := range servers {
+		if server.Spec.Accepted {
+			res = append(res, server)
+		}
+	}
+
+	return res
+}
+
+func filterCPU(servers []Server, filters []CPUInformation) []Server {
```

Comment:
not for this PR, but I think we should refactor this to do one pass through the slice applying all the conditions to avoid copying and iterating over the list many times

Reply:
I will do that in this PR as I already refactored this code